internal/controller/controllers/infraenv_controller.go

```diff
@@ -718,26 +740,37 @@ func generateStaticNetworkConfigDownloadURL(baseURL string, infraEnvId string, a
 
 func (r *InfraEnvReconciler) updateInfraEnvStatus(
 	ctx context.Context, log logrus.FieldLogger, infraEnv *aiv1beta1.InfraEnv, internalInfraEnv *common.InfraEnv) (ctrl.Result, error) {
+	var err error
+	var bootArtifactURLs *imageservice.BootArtifactURLs
+	var isoUpdated bool
 
-	osImage, err := r.OsImages.GetOsImageOrLatest(internalInfraEnv.OpenshiftVersion, internalInfraEnv.CPUArchitecture)
-	if err != nil {
-		return r.handleEnsureISOErrors(ctx, log, infraEnv, err, internalInfraEnv)
-	}
-
-	bootArtifactURLs, err := imageservice.GetBootArtifactURLs(r.ImageServiceBaseURL, internalInfraEnv.ID.String(), osImage, r.InsecureIPXEURLs)
-	if err != nil {
-		return r.handleEnsureISOErrors(ctx, log, infraEnv, err, internalInfraEnv)
+	if infraEnv.Status.InfraEnvDebugInfo.EventsURL == "" {
+		if r.populateEventsURL(log, infraEnv, internalInfraEnv) != nil {
+			return ctrl.Result{Requeue: true}, nil
+		}
 	}
-	infraEnv.Status.BootArtifacts.KernelURL = bootArtifactURLs.KernelURL
-	infraEnv.Status.BootArtifacts.RootfsURL = bootArtifactURLs.RootFSURL
+	// Only update URLs if image service is enabled
+	if r.ImageServiceEnabled {
```

Comment:
Since this entire function is pretty much being rewritten at this point maybe it would be better to have the non-image-service related bits in this function then a whole separate one for the things that need the image service (that might look substantially like what was already here)? That way the diff might be a bit smaller and we could have less nesting in conditionals.